src/main/webapp/gallery.jsp

```diff
@@ -51,6 +51,32 @@ if(request.getParameter("locationCodeField")!=null){
 	locationCodeFieldString="&locationCodeField="+request.getParameter("locationCodeField");
 }
 
+//params from donorbox
+String donorboxId = "";
+if(Util.stringExists(request.getParameter("id"))){
+	donorboxId=request.getParameter("id");
+}
+String donorboxFirstName = "";
+if(Util.stringExists(request.getParameter("first_name"))){
+	donorboxFirstName=request.getParameter("first_name");
+}
+String donorboxLastName = "";
+if(Util.stringExists(request.getParameter("last_name"))){
+	donorboxLastName=request.getParameter("last_name");
+}
+String donorboxAmnt = "";
+if(Util.stringExists(request.getParameter("amount"))){
+	donorboxAmnt=request.getParameter("amount");
+}
+String donorboxCurrency = "";
+if(Util.stringExists(request.getParameter("currency"))){
+	donorboxCurrency=request.getParameter("currency");
+}
+String donorboxDuration = "";
+if(Util.stringExists(request.getParameter("duration"))){
+	donorboxDuration=request.getParameter("duration");
+}
+
```

Comment:
I've also written code that looks like this many times, but we could totally write a better Util method for this purpose when one of us gets around to it.
Something like: 

String someParam = Util.getRequestParamIfExists(request, "paramName");